### PR TITLE
Added telemetry for content buffering

### PR DESCRIPTION
**Pull request by VladyslavAnokhin**
## Changes

Added reporter which works with buffering progress component
Added 3 metric senders which use reporter. Because they have similar logic.

@VerizonAdPlatforms/video-partner-sdk-developers: Please review.

### CODE REVIEW

File: sources/telemetry/Metrics.swift [hunk 1]
```diff
@@ -181,71 +178,104 @@ extension Telemetry.Metrics {
...
+            content.process(playbackSession: state.playbackSession.id,
+                            processingStatus: state.contentBufferingTime.status)
         }
     }
 }
 
-extension Telemetry.Metrics {
-    final class AdBuffering {
+extension Telemetry.Metrics.Buffering {
+    final class MP4Ad {
```
Comment (AndriiMoskvin) [inline sources/telemetry/Metrics.swift]: this can be a struct

File: sources/telemetry/Metrics.swift [hunk 2]
```diff
@@ -181,71 +178,104 @@ extension Telemetry.Metrics {
...
+            vrm.process(uniqueKey: requestId,
+                        processingTime: processingStatus,
+                          telemetryType: "VRM_PROCESSING_TIME")
         }
+    }
+}
+
+extension Telemetry.Metrics.Buffering {
+    final class Content {
+        let content: Reporter
```
Comment (AndriiMoskvin) [inline sources/telemetry/Metrics.swift:268]: this can be a struct

File: sources/telemetry/Metrics.swift [hunk 3]
```diff
@@ -181,71 +178,104 @@ extension Telemetry.Metrics {
...
+            mp4Ad.process(uniqueKey: requestId,
+                          processingTime: processingStatus,
+                          telemetryType: "AD_BUFFERING_TIME")
+        }
+    }
+}
+
+extension Telemetry.Metrics.Buffering {
+    final class VRM {
+        let vrm: Reporter
```
Comment (AndriiMoskvin) [inline sources/telemetry/Metrics.swift:252]: this can be a struct

**Comment by VladyslavAnokhin** [comment]
@AndriiMoskvin I've updated `class` to `struct` where it was possible

**Comment by VladyslavAnokhin** [comment]
@AndriiMoskvin @RomanTysiachnik RFR

**Review by balavor — APPROVED** [review]

**Pull request merged**